homeassistant/components/version/__init__.py

```diff
@@ -18,16 +20,28 @@
 )
 from .coordinator import VersionDataUpdateCoordinator
 
+_LOGGER = logging.getLogger(__name__)
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the version integration from a config entry."""
+
+    board = entry.data[CONF_BOARD]
+
+    if board not in BOARD_MAP:
+        _LOGGER.error(
+            f'Board {board} is (no longer) valid. Removing Version integration "{entry.title}".'
+        )
+        await hass.config_entries.async_remove(entry.entry_id)
```

Comment:
We should not automatically remove it.

Reply:
Hm, didn't you suggest that here? https://github.com/home-assistant/core/pull/72085#discussion_r876088952

Reply:
entries from the dict, not config entries 😄

Reply:
Ah ok, got it now :sweat_smile:. So error and return then.

Reply:
Yeah 👍